File: round-2/web/romance-scam/backend/flag/magic.go
package flag

import (
	"os"
	"strings"

	"github.com/sixafter/nanoid"
)

func GetFlag() string {
	flagTemplate := os.Getenv("FLAG")
	flag := strings.Replace(flagTemplate, "$1", nanoid.MustWithLength(0xb).String(), 1)
	flag = strings.Replace(flag, "$2", nanoid.MustWithLength(0xa).String(), 1)
	flag = strings.Replace(flag, "$3", nanoid.MustWithLength(0xd).String(), 1)
	return flag
}

File: round-1/crypto/not-web/challenge/app/magic.py
import os
import logging
from nanoid import generate

logger = logging.getLogger(__name__)


def generate_flag() -> str:
    try:
        flag_template = os.getenv("FLAG")
        flag = (
            flag_template.replace("$1", generate(size=8))
            .replace("$2", generate(size=8))
            .replace("$3", generate(size=8))
        )
        logger.info(f"Generated flag: {flag}")
        return flag
    except:
        return "Something went terribly wrong, please contact an admin on Discord."
